R/postProcess.R

```diff
@@ -1330,12 +1330,52 @@ postProcessAllSpatial <- function(x, studyArea, rasterToMatch, useCache, filenam
       extRTM <- NULL
       crsRTM <- NULL
     }
+    useBuffer <- FALSE
+    bufferSA <- FALSE
+
+    if (class(x) == "RasterLayer") {
+      #if all CRS are projected, then check if buffer is necessary
+      projections <- sapply(list(x, studyArea, crsRTM), FUN = sf::st_is_longlat)
+      projections <- na.omit(projections)
+      if (!any(projections)) {
+        if (is.null(rasterToMatch)) {
+          useBuffer <- TRUE
+        } else if (max(res(rasterToMatch)) < min(res(x))) {
+          useBuffer <- TRUE
+        }
```

Comment:
both conditions set `useBuffer = TRUE`, so why are we testing in the first place?

Reply:
won't it stay false if there is rasterToMatch but 
```
max(res(rasterToMatch)) > min(res(x)))
```

Reply:
mind you its possible we still end up with the same problem even if the resolution of rasterToMatch is larger than X, especially if its close. I only tested Tati's case, which was 
```
res(x) <- c(10000, 10000) 
res(rasterToMatch) <- c(250, 250)
```
So perhaps I should remove the Ifs and just implement this automatically.

Reply:
can combine into a single condition:

```r
if (is.null(rasterToMatch) || (max(res(rasterToMatch)) < min(res(x))) 
  useBuffer <- TRUE
```

Reply:
ah true. I thought it would throw an error if rasterToMatch was null. I guess not if the first condition is TRUE. I'll make that change.